src/jit/compiler.cpp

```diff
@@ -2730,6 +2730,16 @@ void Compiler::compSetProcessor()
             codeGen->getEmitter()->SetUseSSE4(true);
         }
     }
+#endif
+#if defined(_TARGET_ARM64_)
+// There is no JitFlag for Base instructions handle manually
+#define HARDWARE_INTRINSIC_CLASS_WITH_FLAG
+    opts.setSupportedISA(InstructionSet_Base);
+#define HARDWARE_INTRINSIC_CLASS(flag, isa)                                                                            \
+    if (jitFlags.IsSet(JitFlags::flag))                                                                                \
+        opts.setSupportedISA(InstructionSet_##isa);
+#include "hwintrinsiclistArm64.h"
+
```

Comment:
This sequence is quite confusing. Wouldn't it be clearer just to start with `InstructionSet_Base` as the default?
e.g.:
```
// There is no JitFlag to specify the base instruction set; we start with that as the default.
opts.SetSupportedISA(InstructionSet_Base);
define HARDWARE_INTRINSIC_CLASS(flag, isa)                                                         \
    if (jitFlags.IsSet(JitFlags::flag))                                                            \
        opts.setSupportedISA(InstructionSet_##isa);
#include "hwintrinsiclistArm64.h"

Reply:
:+1: Changing the order of lines would make it clearer.

@CarolEidt Are you also objecting to use of `HARDWARE_INTRINSIC_CLASS_WITH_FLAG` to remove `InstructionSet_Base` from the include list.

Reply:
OK, I think I get it now - you can't omit `InstructionSet_Base` from hwintrinsiclistarm64.h because it's included elsewhere where it will be required. So, you need a way to handle that without the flag. I can't think of a better way to do this off the top of my head, but I definitely think that setting `InstructionSet_Base` first would help in understanding, and then perhaps if you changed `HARDWARE_INTRINSIC_CLASS_WITH_FLAG` to `HARDWARE_INTRINSIC_CLASS_WITH_FLAG_ONLY` it would make it clear that it's a boolean control that ignores the one without a flag.

Reply:
I removed the `HARDWARE_INTRINSIC_CLASS_WITH_FLAG` and always handle base manually.

It is simpler and easier to understand.